okio-nodefilesystem/src/main/kotlin/okio/NodeJsFileSystem.kt

```diff
@@ -30,27 +30,39 @@ object NodeJsFileSystem : FileSystem() {
   private var S_IFMT = 0xf000 // fs.constants.S_IFMT
   private var S_IFREG = 0x8000 // fs.constants.S_IFREG
   private var S_IFDIR = 0x4000 // fs.constants.S_IFDIR
+  private var S_IFLNK = 0xa000 // fs.constants.S_IFLNK
 
   override fun canonicalize(path: Path): Path {
     try {
       val canonicalPath = realpathSync(path.toString())
-      return canonicalPath.toString().toPath()
+      return canonicalPath.toPath()
     } catch (e: Throwable) {
       throw e.toIOException()
     }
   }
 
   override fun metadataOrNull(path: Path): FileMetadata? {
+    val pathString = path.toString()
     val stat = try {
-      statSync(path.toString())
+      lstatSync(pathString)
     } catch (e: Throwable) {
       if (e.errorCode == "ENOENT") return null // "No such file or directory".
       throw IOException(e.message)
     }
+
+    var symlinkTarget: Path? = null
+    if ((stat.mode.toInt() and S_IFMT) == S_IFLNK) {
+      try {
+        symlinkTarget = readlinkSync(pathString).toPath()
+      } catch (e: Throwable) {
+        throw e.toIOException()
```

Comment:
there’s an ugly race where the file is a symlink on the 1st call and not on the 2nd call. I don’t think we need to do anything about it.